pkgs/development/libraries/ngtcp2/default.nix

```diff
@@ -15,8 +16,9 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-uBmD26EYT8zxmHD5FuHCbEuTdWxer/3uhRp8PhUT87M=";
   };
 
-  nativeBuildInputs = [ autoreconfHook pkg-config cunit file ];
-  buildInputs = [ jemalloc libev nghttp3 quictls ];
+  nativeBuildInputs = [ autoreconfHook pkg-config file ];
+  buildInputs = [ libev nghttp3 quictls ] ++ lib.optional withJemalloc jemalloc;
```

Comment:
Maybe this an variant?
```
  buildInputs = [ libev nghttp3 quictls ] ++ lib.optional stdenv.hostPlatform.isLinux jemalloc;
```

Reply:
I disabled jemalloc by default because in addition to the things that I mentioned in https://github.com/NixOS/nixpkgs/pull/158008#issuecomment-1029272051, I believe that for this particular case, the cons of enabling it likely outweigh the pros. 

Currently, the only consumer of ngtcp2 is curl. When you consider the command line use of curl, the primary bottleneck is going to be network speed. I doubt replacing the memory allocator would result in significant improvements. It's also going to be a short-lived process with one or two threads, so jemalloc selling points like fragmentation avoidance and better concurrency support isn't really a concern. For short-lived processes, the additional cost of loading and setting up an additional memory allocator might even exceed any performance gains if there are any.

Even if you consider long running processes stemming from the use of libcurl, enabling jemalloc would likely not help here too. Because in order for jemalloc to be effective at what it does, you'd need all (or most) parts your program to stick with jemalloc. However, this would not be the case for curl because jemalloc would only be used for ngtcp2. curl itself and all other libraries would still use glibc's malloc, defeating the point.

Considering the increased dependency closures, inconsistency between platforms caused by conditional inclusion, slightly increased runtime initialization overhead, and the risk of using a less common setup, I think it's better to have it disabled by default at least for now.